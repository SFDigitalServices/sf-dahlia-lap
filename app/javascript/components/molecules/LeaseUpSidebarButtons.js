import React from 'react'
import PropTypes from 'prop-types'

import StatusDropdown from './StatusDropdown'

import { LEASE_UP_STATUS_VALUES } from '../../utils/statusUtils'

const LeaseUpSidebarButtons = ({ status, withMobileStyling, isLoading, onSaveClicked, onChangeStatus, onAddCommentClicked }) => {
  const saveButton = (
    <button
      className='primary save-button'
      id='save-supplemental-application'
      type='submit'
      disabled={isLoading}
      onClick={onSaveClicked}
    >
      {isLoading ? 'Savingâ€¦' : 'Save'}
    </button>
  )

  const statusHistoryButtons = (
    <div className='status-history-buttons'>
      <StatusDropdown
        status={status}
        onChange={onChangeStatus}
        disabled={isLoading}
        buttonClasses={['tight-padding']}
      />
      <button
        // ID is just for e2e tests
        id='add-status-history-comment'
        className='tertiary tight-padding'
        disabled={isLoading}
        onClick={onAddCommentClicked}
      >
        Add a comment
      </button>
    </div>
  )

  return (
    <>
      {!withMobileStyling && saveButton}
      {statusHistoryButtons}
      {withMobileStyling && saveButton}
    </>
  )
}

LeaseUpSidebarButtons.propTypes = {
  status: PropTypes.oneOf(LEASE_UP_STATUS_VALUES).isRequired,
  withMobileStyling: PropTypes.bool,
  isLoading: PropTypes.bool,
  onSaveClicked: PropTypes.func,
  onChangeStatus: PropTypes.func,
  onAddCommentClicked: PropTypes.func
}

LeaseUpSidebarButtons.defaultProps = {
  withMobileStyling: false,
  isLoading: false,
  onSaveClicked: null,
  onChangeStatus: null,
  onAddCommentClicked: null
}

export default LeaseUpSidebarButtons
