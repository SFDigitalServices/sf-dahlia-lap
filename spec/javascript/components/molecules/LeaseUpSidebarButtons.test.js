import React from 'react'
import { shallow } from 'enzyme'
import LeaseUpSidebarButtons from 'components/molecules/LeaseUpSidebarButtons'

describe('LeaseUpSidebarButtons', () => {
  test('should render correctly with mobile styling', () => {
    const wrapper = shallow(
      <LeaseUpSidebarButtons
        status={'Approved'}
        withMobileStyling
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('should render correctly with desktop styling', () => {
    const wrapper = shallow(
      <LeaseUpSidebarButtons
        status={'Approved'}
        withMobileStyling={false}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('should render desktop styling by default', () => {
    const wrapper = shallow(
      <LeaseUpSidebarButtons
        status={'Approved'}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
