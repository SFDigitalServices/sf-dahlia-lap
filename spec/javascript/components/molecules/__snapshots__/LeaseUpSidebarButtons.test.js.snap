// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaseUpSidebarButtons should render correctly with desktop styling 1`] = `
<Fragment>
  <button
    className="primary save-button"
    onClick={null}
  >
    Save
  </button>
  <div
    className="status-history-buttons"
  >
    <StatusDropdown
      buttonClasses={
        Array [
          "tight-padding",
        ]
      }
      onChange={null}
      status="Approved"
    />
    <button
      className="tertiary tight-padding"
      onClick={null}
    >
      Add a comment
    </button>
  </div>
</Fragment>
`;

exports[`LeaseUpSidebarButtons should render correctly with mobile styling 1`] = `
<Fragment>
  <div
    className="status-history-buttons"
  >
    <StatusDropdown
      buttonClasses={
        Array [
          "tight-padding",
        ]
      }
      onChange={null}
      status="Approved"
    />
    <button
      className="tertiary tight-padding"
      onClick={null}
    >
      Add a comment
    </button>
  </div>
  <button
    className="primary save-button"
    onClick={null}
  >
    Save
  </button>
</Fragment>
`;

exports[`LeaseUpSidebarButtons should render desktop styling by default 1`] = `
<Fragment>
  <button
    className="primary save-button"
    onClick={null}
  >
    Save
  </button>
  <div
    className="status-history-buttons"
  >
    <StatusDropdown
      buttonClasses={
        Array [
          "tight-padding",
        ]
      }
      onChange={null}
      status="Approved"
    />
    <button
      className="tertiary tight-padding"
      onClick={null}
    >
      Add a comment
    </button>
  </div>
</Fragment>
`;
